src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs

```diff
@@ -155,25 +152,66 @@ public override TPredictor Train(TrainContext context)
 
         public override PredictionKind PredictionKind => PredictionKind.BinaryClassification;
 
-        public SymSgdClassificationTrainer(IHostEnvironment env, Arguments args)
-            : base(env, LoadNameValue)
+        /// <summary>
+        /// Initializes a new instance of <see cref="SymSgdClassificationTrainer"/>
+        /// </summary>
+        /// <param name="env">The private instance of <see cref="IHostEnvironment"/>.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="featureColumn">The name of the feature column.</param>
+        /// <param name="weightColumn">The name for the column containing the initial weight.</param>
+        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
+        public SymSgdClassificationTrainer(IHostEnvironment env, string featureColumn, string labelColumn,
+            string weightColumn = null, Action<Arguments> advancedSettings = null)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(LoadNameValue), TrainerUtils.MakeR4VecFeature(featureColumn),
+                  TrainerUtils.MakeBoolScalarLabel(labelColumn), TrainerUtils.MakeR4ScalarWeightColumn(weightColumn))
+        {
+            _args = new Arguments();
+
+            // Apply the advanced args, if the user supplied any.
+            _args.Check(Host);
+            advancedSettings?.Invoke(_args);
+            _args.FeatureColumn = featureColumn;
```

Comment:
>args.FeatureColumn [](start = 13, length = 18)

weights column

Reply:
Still missing `weightColumn` a couple commits after @sfilipi's comment

---
In reply to: [220300485](https://github.com/dotnet/machinelearning/pull/1012#discussion_r220300485) [](ancestors = 220300485)